filesystem/trd.ksy

```diff
@@ -0,0 +1,138 @@
+meta:
+  id: trd
+  file-extension: trd
+  title: "TR-DOS flat-file disk image"
+  license: CC0-1.0
+  # It's extended ascii (with ZX Spectrum Basic tokens, UDG, etc), but
+  # "unknown-8bit" can't be used for some reason
+  encoding: ascii
+  endian: le
+doc: |
+  .trd file is a raw dump of TR-DOS (ZX-Spectrum) floppy. .trd files are
+  headerless and contain consequent "logical tracks", each logical track
+  consists of 16 256-byte sectors.
+
+  Logical tracks are defined the same way as used by TR-DOS: for single-side
+  floppies it's just a physical track number, for two-side floppies sides are
+  interleaved, i.e. logical_track_num = (physical_track_num << 1) | side
+
+  So, this format definition is more for TR-DOS filesystem than for .trd files,
+  which are formatless.
+
+  Strings (file names, disk label, disk password) are padded with spaces and use
+  ZX Spectrum character set, including UDGs, block drawing chars and Basic
+  tokens. ASCII range is mostly standard ASCII, with few characters (^, `, DEL)
+  replaced with (up arrow, pound, copyright symbol).
+seq:
+  - id: files
+    type: file
+    repeat: until
+    # After 128 files there is disk info entry, which also has 0x00 terminator
+    # in the same position as file name. So usually even with 128 files you can
+    # just read until 0x00.
+    repeat-until: _.is_terminator
+instances:
+  volume_info:
+    pos: 0x800
+    type: volume_info
+types:
+  file:
+    seq:
+      - id: name
+        type: str
+        size: 8
+      - id: extension
+        type: str
+        size: 1
+
+      - id: program_and_data_length
+        type: u2
+        if: extension == "B"
+      - id: start_address
+        doc: Default memory address to load this byte array into
+        type: u2
+        if: extension == "C"
+      - id: print_extent_num
+        type: u1
+        if: extension == "#"
+      - id: print_unused
+        type: u1
+        if: extension == "#"
+      - id: length1_unused
+        size: 2
+        if: extension != "B" and extension != "C" and extension != "#"
```

Comment:
A suggestion: why not move this tons of `if` into separate types and just use type switching (i.e. `switch-on: extension`)?

Reply:
Thanks, it's easier and cleaner with `switch-on`. I didn't know about this feature because it's not listed in KSY reference, only in User guide.
Changed to `switch-on`.